src/deps-index.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {Analyzer} from 'polymer-analyzer';
+
+export interface DepsIndex {
+  // An index of dependency -> fragments that depend on it
+  depsToFragments: Map<string, Set<string>>;
+  // An index of fragments -> html dependencies
+  fragmentToDeps: Map<string, Set<string>>;
```

Comment:
The only kind of map which matters is this fragmentToDeps one since we have a function to produce a set of initial bundle definitions based purely on it.  The depsToFragments is ephemeral and only really used when performing this initial bundle set construction.